cdc/processor/sinkmanager/mem_quota.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sinkmanager
+
+import "github.com/pingcap/tiflow/cdc/model"
+
+//nolint:varcheck,deadcode
+const defaultMemoryUsage = 10 * 1024 * 1024 // 10MB
+
+type memQuota interface {
+	// TryAcquire try to acquire n bytes of memory.
+	TryAcquire() bool
+	// ForceAcquire force to acquire n bytes of memory.
+	// It will exceed the total memory limit.
+	ForceAcquire()
```

Comment:
Seems we need
```go
func ForceAcquire(clientID int, bytes int)
```

`MemoryQuota` should ensure there is at most 1 client can acquire memory exceeds the limit.